server/src/main/java/org/opensearch/cluster/routing/WeightedRoutingService.java

```diff
@@ -154,4 +158,15 @@ public void verifyAwarenessAttribute(String attributeName) {
             throw validationException;
         }
     }
+
+    public void ensureNoOngoingDecommissionAction(ClusterState state) {
+        DecommissionAttributeMetadata decommissionAttributeMetadata = state.metadata().decommissionAttributeMetadata();
+        if (decommissionAttributeMetadata != null && decommissionAttributeMetadata.status().equals(DecommissionStatus.FAILED) == false) {
```

Comment:
If decommission status is FAILED can certain nodes be out of service or decommissioned while others not as in partially failed?

Reply:
No, if the decommission is `FAILED`, joins or prevoting is not blocked on those nodes and hence if some nodes are out of service (which can happen), the weights request can go through

Reply:
But then the zone isn't fully equipped to take traffic right like 90% nodes are out of service and we place weights 10:1:1 when 10 is for the zone only operating at 10% capacity?

Reply:
No